### ISSUE TITLE
Suggestion: Merge screenshots feature

### ISSUE THREAD

**Issue by Pedrotheplant** [nikobojs/manuscrape_electron#70]
Can we include a merge screenshots function, so that we can take a screenshot, and then take another and indicate on what side of the original shot we want it merged?

On some platforms, e.g., instagram, the comments are in a separate scrollable field, meaning that you can't screenshot both the main image and the comments beyond what is immediately visible.

This could be solved by first screenshotting what is immediately visible, and then screenshotting all the other comments and merge them onto the left side of the image. It does not look nice but for analytical purposes it is more than ok.

**Comment by nikobojs** [comment]
In your example you have 2 image-files - the image and the comments. Because manuscrape has one primary picture for upload, you want to merge these two pictures into one. Right?

If this is the case - maybe the constraint having only one primary picture pr. observation is the actual problem.

Is there any way it could make sense for you to upload these images separately ?

**Comment by Pedrotheplant** [comment]
That is correctly understood! Uploading them seperately would definitely be a quick fix that could work, however analytically it would be nice to have them in one single image. This would also lower the chances of accidentally not including the second image when analyzing after export.

**Comment by nikobojs** [comment]
Interesting! I think we need to step back and think a little bit - there might be bigger things to decide before we can actually solve this. And I still don't think I understand the value proposition correctly.

1. What makes it analytically nice to have 2 images with different content in one image?
_My thoughts: It would be more structured, less fiddling of observation data, easier to export/find what you want, and in many ways just better with 2 pictures than 1. All around flexibility and structure would decrease if we join images, just like joining 2 columns in excel with different types and meanings._

2. How would it lower the chances of accidentally not including a second image?
_My thoughts: What if there were 2 required image upload fields? (I know manuscrape does not support this yet) - This would lower the chances. If you have multiple pictures in one, you open pandoras box of human mistakes as you wont know if the image is joined correctly (following the project setup) or not._

For instance, if your project has observations that needs "Comments image" and "Main image" - wouldn't it be better to upload these separately? It would allow making comments optional but main image required. While keeping flexibility for adding different helper/texts and future rules for the different images. It would also make sense if you could pick which columns and/or files you export. In this case joined observation image data could be a very limiting constraint.

Because the "Comments" and "Main image" conceptually are different things, I believe they must also be separate in technical terms, to make most sense for most people. Just like you wouldn't merge two excel columns into one, without loosing data-quality and making stuff more difficult to understand.

A feature like this is a relatively big task. Not because of image joining, but because of all the user-experience work that is required. We'll need to actually create new images that can import other images.

It smells a bit like feature creep so just trying to take a step back and think about what we're doing. My gut feeling tells me that there is a UX problem that makes the end user want "anti-features". Not because you're wrong or anything!

I think we need to rethink the one-primary-image-per-observation rule the application is based upon.

Rambling a bit here, I hope it makes sense. Any thoughts?

**Comment by Pedrotheplant** [comment]
Great thoughts, makes a lot of sense! Let me try to expand a little bit:

1. A part of analyzing is being able to have the "full picture" in front of you, meaning that you can see both main post and comments at the same time. Experentially there is quite a big difference between this and then clicking back and forth between multiple screenshots each containing a small amount of the big picture. 

2. I do not have a good answer to this very relevant question. Hitherto we've used ShareX's image combiner-feature, where you get to check the merge before saving it. However, I definitely see the problems you're mentioning, however at the same time, not including this feature might also be pushing the same problem further down the road. E.g., in a screenshot with 5 separate pictures, what are the chances that one of these will be missed when analyzing, or read in the wrong order?

However, I definitely see your points, both regarding validity and feature creep, so let's hold back on this one for now.
